source/extensions/filters/http/aws_request_signing/config.cc

```diff
@@ -25,11 +46,22 @@ Http::FilterFactoryCb AwsRequestSigningFilterFactory::createFilterFactoryFromPro
       std::make_shared<Extensions::Common::Aws::DefaultCredentialsProviderChain>(
           server_context.api(), makeOptRef(server_context), config.region(),
           Extensions::Common::Aws::Utility::fetchMetadata);
-  const auto matcher_config = Extensions::Common::Aws::AwsSigV4HeaderExclusionVector(
+  const auto matcher_config = Extensions::Common::Aws::AwsSigningHeaderExclusionVector(
       config.match_excluded_headers().begin(), config.match_excluded_headers().end());
-  auto signer = std::make_unique<Extensions::Common::Aws::SignerImpl>(
-      config.service_name(), config.region(), credentials_provider,
-      server_context.mainThreadDispatcher().timeSource(), matcher_config);
+
+  std::unique_ptr<Extensions::Common::Aws::Signer> signer;
+  Envoy::Logger::Registry::setLogLevel(spdlog::level::debug);
+
+  if (getSigningAlgorithm(config) == SigningAlgorithm::SIGV4A) {
+    signer = std::make_unique<Extensions::Common::Aws::SigV4ASignerImpl>(
+        config.service_name(), config.region(), credentials_provider,
+        server_context.mainThreadDispatcher().timeSource(), matcher_config);
+  } else {
+    signer = std::make_unique<Extensions::Common::Aws::SigV4SignerImpl>(
```

Comment:
can we throw Envoy exception here if the region contains wildcard (`*`) or region set (`,`) ?

Reply:
as above